src/mono/mono/metadata/marshal.c

```diff
@@ -3711,7 +3711,7 @@ mono_marshal_get_native_wrapper (MonoMethod *method, gboolean check_exceptions,
 						swift_error_args++;
 					} else if (param_klass == swift_self) {
 						swift_self_args++;
-					} else if (!m_class_is_blittable (param_klass) || m_class_is_simd_type (param_klass)) {
```

Comment:
During AOT, class might not be properly initialize thus causing this check to fail even for blittable types.

Reply:
Can you call `mono_class_init_internal (mono_class_from_mono_type (method->signature->params[i]))` before checking if the type is blittable?

Reply:
The issue with AOT was that the wrappers don't get llvm compiled but are JIT/interpreter compiled during runtime and the blittable field was not correctly set at the time of this check. I believe calling class init wouldn't solve it, we would have to use `mono_class_setup_fields` (I *think*, haven't tried it). Do you think `mono_class_setup_fields (mono_class_from_mono_type (method->signature->params[i]))` would be better than just using `type_is_blittable (method->signature->params [i])`?

(I put the comment just to explain why I did the change, the code change in this PR makes it work with all codegens).

Reply:
yea try calling setup_fields before the blittable check

Reply:
I misunderstood the diff.  and I see `type_is_blittable` already calls `mono_class_init_sizes` which sets up fields when necessary.